### PR TITLE
fix: remove access points on unlink installation

### PR DESCRIPTION
**Pull request by gcusnieux**
## Issue

https://gravitee.atlassian.net/browse/APIM-3590

## Description

A small description of what you did in that PR.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-wboliyosgr.chromatic.com)

### CODE REVIEW

File: gravitee-apim-repository/gravitee-apim-repository-test/src/test/java/io/gravitee/repository/management/OrganizationRepositoryTest.java [hunk 1]
```diff
@@ -119,4 +119,10 @@ public void shouldFindByHrids() throws Exception {
         Assert.assertTrue("No organization found", !organizations.isEmpty());
         Assert.assertEquals(2, organizations.size());
     }
+
+    @Test
+    public void shouldFindByCockpitId() throws Exception {
+        final Optional<Organization> optional = organizationRepository.findByCockpit("cockpitId-org-findById");
+        Assert.assertTrue("Organization to find not found", optional.isPresent());
```
Comment (ytvnr) [inline gravitee-apim-repository/gravitee-apim-repository-test/src/test/java/io/gravitee/repository/management/OrganizationRepositoryTest.java]:
```suggestion
        Assert.assertTrue("No organization found for cockpitId: cockpitId-org-findById", optional.isPresent());
```

File: gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/OrganizationRepository.java [hunk 2]
```diff
@@ -26,4 +27,5 @@
 public interface OrganizationRepository extends CrudRepository<Organization, String> {
     Long count() throws TechnicalException;
     Set<Organization> findByHrids(Set<String> hrids) throws TechnicalException;
+    Optional<Organization> findByCockpit(String cockpitId) throws TechnicalException;
```
Comment (ytvnr) [inline gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/OrganizationRepository.java]: in the service and mongo implem we have `findByCockpitId`, should we keep the same naming ?

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcOrganizationRepository.java [hunk 3]
```diff
@@ -126,6 +127,29 @@ public Set<Organization> findByHrids(Set<String> hrids) throws TechnicalExceptio
...
+    public Optional<Organization> findByCockpit(String cockpitId) throws TechnicalException {
+        LOGGER.debug("JdbcOrganizationRepository.findByCockpit({})", cockpitId);
+        try {
+            List<Organization> organizations = jdbcTemplate.query(
+                getOrm().getSelectAllSql() + " where cockpit_id = ?",
+                getOrm().getRowMapper(),
+                cockpitId
+            );
+
+            final Optional<Organization> organization = organizations.stream().findFirst();
```
Comment (ytvnr) [inline gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcOrganizationRepository.java:140]: maybe I miss something, but why taking the first only, for a cockpit ID we can have multiple orga (or env) no ? 
So how to unlink everything for a cockpit id ? 

EDIT: from tests, I see the cockpitId are containing info about env or org, if it is the case in real life then no problem to find the first

Comment (jgiovaresco) [reply gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcOrganizationRepository.java:140]: There is a cockpitId for organization and a cockpitId for environment. When the organization/environment is created through cockpit, the cockpitId and _id is the same. However for the default organization/environment, the _id is DEFAULT and when the user link an installation to the default env, the command will update the cockpitId in APIM.

So basically, the cockpitId should be unique

File: gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/OrganizationRepository.java (hunk 2 above)
Comment (gcusnieux) [reply gravitee-apim-repository/gravitee-apim-repository-api/src/main/java/io/gravitee/repository/management/api/OrganizationRepository.java]: Ok, but I will refactor EnvironmentService also ? (Its the same naming)

File: gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcOrganizationRepository.java (hunk 3 above)
Comment (gcusnieux) [reply gravitee-apim-repository/gravitee-apim-repository-jdbc/src/main/java/io/gravitee/repository/jdbc/management/JdbcOrganizationRepository.java:140]: Good point ! 
I agree this part is strange, but we have the same implementation In EnvRepo... 
And in fact, with the test describe in https://gravitee.atlassian.net/browse/APIM-3590 we have no unique cockpitId in collection... @jgiovaresco do you have time to discuss on this part today ?

**Review by ytvnr — APPROVED** [review]
lgtm

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/EnvironmentCommandHandler.java [hunk 4]
```diff
@@ -99,13 +101,17 @@ public Single<EnvironmentReply> handle(EnvironmentCommand command) {
...
+            log.error("Error occurred when handling environment [{}] with id [{}].", environmentPayload.getName(), environmentId, e);
             return Single.just(new EnvironmentReply(command.getId(), CommandStatus.ERROR));
         }
     }
+
+    private String getEnvironmentId(EnvironmentPayload environmentPayload) {
+        try {
+            EnvironmentEntity environment = this.environmentService.findByCockpitId(environmentPayload.getCockpitId());
+            return environment.getId();
+        } catch (Exception ex) {
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/EnvironmentCommandHandler.java]:
```suggestion
        } catch (EnvironmentNotFoundException ex) {
```

I would only use the payload id if the environment is not found if a technical error occurs we could create a duplicate 🤔

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/OrganizationCommandHandler.java [hunk 5]
```diff
@@ -95,13 +98,17 @@ public Single<OrganizationReply> handle(OrganizationCommand command) {
...
+            log.error("Error occurred when handling organization [{}] with id [{}].", organizationPayload.getName(), organizationId, e);
             return Single.just(new OrganizationReply(command.getId(), CommandStatus.ERROR));
         }
     }
+
+    private String getOrganizationId(OrganizationPayload organizationPayload) {
+        try {
+            OrganizationEntity byCockpitId = this.organizationService.findByCockpitId(organizationPayload.getCockpitId());
+            return byCockpitId.getId();
+        } catch (Exception ex) {
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/OrganizationCommandHandler.java]:
```suggestion
        } catch (OrganizationNotFoundException ex) {
```
I would only use the payload id if the organization is not found if a technical error occurs we could create a duplicate 🤔

File: gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/EnvironmentCommandHandler.java [hunk 6]
```diff
@@ -63,7 +65,7 @@ public Single<EnvironmentReply> handle(EnvironmentCommand command) {
 
             final EnvironmentEntity environment = environmentService.createOrUpdate(
                 environmentPayload.getOrganizationId(),
```
Comment (jgiovaresco) [inline gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/cockpit/command/handler/EnvironmentCommandHandler.java]: I think we should use the organizationId of the existing environment if it exists 🤔

**Review by jgiovaresco — CHANGES_REQUESTED** [review]

**Review by jgiovaresco — APPROVED** [review]

**Pull request merged**